### PR TITLE
Add support for nxOMSAuditdPlugin telemetry

### PR DESCRIPTION
**Pull request by taglines**

### CODE REVIEW

**Review by sethRait — APPROVED** [review]
LGTM

File: source/code/plugins/filter_auditd_plugin.rb [hunk 1]
```diff
@@ -32,4 +32,36 @@ def filter(tag, time, record)
         return audit_blob
     end # filter
   end # class
+
+  class AuditdDSCLogFilter < Filter
+
+    Fluent::Plugin.register_filter('filter_auditd_dsc_log', self)
```
Comment (hestolz) [inline source/code/plugins/filter_auditd_plugin.rb:38]: where's the `filter_auditd_dsc_log` being used? don't see in any .conf

File: source/code/plugins/auditd_plugin_lib.rb [hunk 2]
```diff
@@ -63,4 +64,72 @@ def transform_and_wrap(event, hostname, time)
...
+            dataitem["version"] = dsc_version
+            dataitem["Message"] = operation_result
+            dataitem["Data"] = {
+                "has_auditd"=>has_auditd,
+                "auditd_pid"=>auditd_pid,
+                "auomscollect_pid"=>auomscollect_pid,
+                "auoms_pid"=>auoms_pid
+            }.to_json
+
+            @log.info "nxOMSAuditdPlugin STATUS: " + dataitem.to_s
```
Comment (hestolz) [inline source/code/plugins/auditd_plugin_lib.rb:114]: how often does this execute? ideally far less than 1x/min

**Review by hestolz — APPROVED** [review]

File: source/code/plugins/filter_auditd_plugin.rb (hunk 1 above)
Comment (taglines) [reply source/code/plugins/filter_auditd_plugin.rb:38]: It is used in the auditd_plugin.conf. See: dsc/Providers/Modules/nxOMSAuditdPlugin/conf/auditd_plugin.conf
Here is the relevent PR in the dsc repo: https://github.com/microsoft/PowerShell-DSC-for-Linux/pull/758

File: source/code/plugins/auditd_plugin_lib.rb (hunk 2 above)
Comment (taglines) [reply source/code/plugins/auditd_plugin_lib.rb:114]: The DSC module runs once every 15 minutes (driven by OMSConsistencyInvoker cron job). And there are two messages per Op (Test & Set), so typically, 4 per 15-minutes.

**Pull request merged**